packages/x-data-grid-pro/src/components/GridServerSideTreeDataGroupingCell.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { unstable_composeClasses as composeClasses } from '@mui/utils';
+import { styled } from '@mui/system';
+import Box from '@mui/material/Box';
+import Badge from '@mui/material/Badge';
+import {
+  getDataGridUtilityClass,
+  GridRenderCellParams,
+  GridServerSideGroupNode,
+  useGridSelector,
+} from '@mui/x-data-grid';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useGridRootProps } from '../hooks/utils/useGridRootProps';
+import { useGridPrivateApiContext } from '../hooks/utils/useGridPrivateApiContext';
+import { DataGridProProcessedProps } from '../models/dataGridProProps';
+import { GridPrivateApiPro } from '../models/gridApiPro';
+import { GridStatePro } from '../models/gridStatePro';
+
+type OwnerState = { classes: DataGridProProcessedProps['classes'] };
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['treeDataGroupingCell'],
+    toggle: ['treeDataGroupingCellToggle'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+interface GridTreeDataGroupingCellProps
+  extends GridRenderCellParams<any, any, any, GridServerSideGroupNode> {
+  hideDescendantCount?: boolean;
+  /**
+   * The cell offset multiplier used for calculating cell offset (`rowNode.depth * offsetMultiplier` px).
+   * @default 2
+   */
+  offsetMultiplier?: number;
+}
+
+interface GridTreeDataGroupingCellIconProps
+  extends Pick<GridTreeDataGroupingCellProps, 'id' | 'field' | 'rowNode' | 'row'> {
+  descendantCount: number;
+}
+
+const LoadingContainer = styled('div')({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  height: '100%',
+});
```

Comment:
For the time being as we still use emotion this is not good for performance. If there can be a lot of loading containers mounted at once, this could have an impact.

Reply:
Although it's not likely that a lot of such components would be mounted at once, except with auto fetching using `defaultGroupingExpansionDepth` for example. Do you think moving this style to root styles would reduce the performance cost a bit?

Reply:
Yes, that would be the best place.